elastica/boundary_conditions.py

```diff
@@ -152,7 +223,178 @@ def compute_constrain_rates(velocity_collection, omega_collection):
         omega_collection[..., 0] = 0.0
 
 
-class HelicalBucklingBC(FreeRod):
+class OneEndFixedRod(OneEndFixedBC):
+    # Please clear this part beyond version 0.3.0
+    """Deprecated 0.2.1: Same implementation as OneEndFixedBC"""
+    warnings.warn(
+        "OneEndFixedRod is deprecated and renamed to OneEndFixedBC. The deprecated name will be removed in the future.",
+        DeprecationWarning,
+    )
+
+
+class FixedConstraint(ConstraintBase):
+    """
+    This boundary condition class fixes the specified node or orientations.
+    Index can be passed to fix either or both the position or the director.
+    Constraining position is equivalent to setting 0 translational DOF.
+    Constraining director is equivalent to setting 0 rotational DOF.
+
+    Examples
+    --------
+    How to fix two ends of the rod:
+
+    >>> simulator.constrain(rod).using(
+    ...    FixedConstraint,
+    ...    constrained_position_idx=(0,1,-2,-1),
+    ...    constrained_director_idx=(0,-1)
+    ... )
+
+    How to pin the middle of the rod (10th node), without constraining the rotational DOF.
+
+    >>> simulator.constrain(rod).using(
+    ...    FixedConstraint,
+    ...    constrained_position_idx=(10)
+    ... )
+    """
+
+    def __init__(self, *fixed_data, **kwargs):
+        """
+
+        Parameters
+        ----------
+        fixed_data : tuple
+            Tuple of position and directors
+        """
+        super().__init__(**kwargs)
+        pos, dir = [], []
+        for data in fixed_data:
+            if isinstance(data, np.ndarray) and data.shape == (3,):
+                pos.append(data)
+            elif isinstance(data, np.ndarray) and data.shape == (
+                3,
+                3,
+            ):
+                dir.append(data)
+            else:
+                # TODO: This part is prone to error.
+                break
+        self.fixed_positions = np.array(pos)
+        self.fixed_directors = np.array(dir)
```

Comment:
An idea maybe is  hiding all these from user and handle everything inside the `_Constraint` class.

Reply:
Other functions' `__init__` include saving initial position/directors (`HelicalBucklingBC`, `OneEndFixedBC`). If we want to include this step in all constraints, we can add this inside the `_Constraint` class, but that would need a change in all implementations. Maybe we can refactor that in ` v0.3.0`. I wanted to keep this update `v0.2.1 -> v0.2.2`  to be soft.

Reply:
Okay then lets target v.0.3.0 for that.